bot/exts/help_channels/_cog.py

```diff
@@ -141,3 +143,17 @@ async def new_post_message_listener(self, message: discord.Message) -> None:
 
         if not message.author.bot and message.author.id != message.channel.owner_id:
             await _caches.posts_with_non_claimant_messages.set(message.channel.id, "sentinel")
+
+    @commands.Cog.listener()
+    async def on_member_remove(self, member: discord.Member) -> None:
+        """Notify a help thread if the owner is no longer a member of the server."""
+        for thread in self.help_forum_channel.threads:
+            if thread.owner_id != member.id:
+                continue
+
+            if thread.archived:
+                continue
+
+            log.debug(f"Notifying help thread {thread.id} that owner {member.id} has left the server.")
+            with contextlib.suppress(discord.NotFound):
+                await thread.send(":warning: The owner of this post has left the server.")
```

Comment:
A slight nit - the rest looks good as a code review 👍 

Looking at the one instance this was needed from the linked issue, it may not always (or even likely) be the owner deliberately leaving the server rather than getting a ban in such cases. How's this instead?
```suggestion
                await thread.send(":warning: The owner of this post is no longer in the server.")
```

Reply:
Yeah, that's better 👍

Reply:
Actually i'll change the log above too so it matches.

Reply:
done